ansible_events_ui/api/activation.py

```diff
@@ -45,3 +45,97 @@ async def create_activation(
+    "/api/activation/{activation_id}",
+    response_model=schemas.ActivationRead,
+    operation_id="show_activation",
+)
+async def read_activation(
+    activation_id: int, db: AsyncSession = Depends(get_db_session)
+):
+    query = (
+        sa.select(
+            models.activations.c.id,
+            models.activations.c.name,
+            models.activations.c.description,
+            models.activations.c.activation_enabled,
+            models.activations.c.activation_status,
+            models.activations.c.restarted_at,
+            models.activations.c.restarted_count,
+            models.activations.c.created_at,
+            models.activations.c.modified_at,
+            models.rulebooks.c.id.label("rulebook_id"),
+            models.rulebooks.c.name.label("rulebook_name"),
+            models.inventories.c.id.label("inventory_id"),
+            models.inventories.c.name.label("inventory_name"),
+            models.extra_vars.c.id.label("extra_var_id"),
+            models.extra_vars.c.name.label("extra_var_name"),
+            models.playbooks.c.id.label("playbook_id"),
+            models.playbooks.c.name.label("playbook_name"),
+            models.restart_policies.c.id.label("restart_policy_id"),
+            models.restart_policies.c.name.label("restart_policy_name"),
+            models.execution_envs.c.id.label("execution_env_id"),
+            models.execution_envs.c.url.label("execution_env_url"),
+        )
+        .select_from(
+            models.activations.join(models.rulebooks)
+            .join(models.inventories)
+            .join(models.extra_vars)
+            .join(models.playbooks)
+            .join(models.restart_policies)
+            .join(models.execution_envs)
+        )
+        .where(models.activations.c.id == activation_id)
+    )
+    result = (await db.execute(query)).one_or_none()
+    if result is None:
+        raise HTTPException(status_code=404, detail="Activation Not Found.")
+    return result
+
+
+@router.patch(
+    "/api/activation/{activation_id}",
+    response_model=schemas.ActivationCreate,
+    operation_id="update_activation",
+)
+async def update_activation(
+    activation_id: int,
+    activation: schemas.ActivationUpdate,
+    db: AsyncSession = Depends(get_db_session),
+):
+    stored_activation = (
+        await db.execute(
+            sa.select(models.activations).where(
+                models.activations.c.id == activation_id
+            )
+        )
+    ).one_or_none()
+    if stored_activation is None:
+        raise HTTPException(status_code=404, detail="Activation Not Found.")
+
+    try:
+        await db.execute(
+            sa.update(models.activations)
+            .where(models.activations.c.id == activation_id)
+            .values(
+                name=activation.name,
+                description=activation.description,
+                activation_enabled=activation.activation_enabled,
+            )
+        )
+    except sa.exc.IntegrityError:
```

Comment:
The `sqlalchemy.exc` import is missing. The `sqlalchemy` package doesn't import the `exc` module. You should avoid relying on implicit imports.

Reply:
In which particular case you expect `IntegrityError` raised here?

Reply:
I'm re-using the `except` clause on line 42 in this file which you've suggested yourself.

Reply:
I was using `IntegrityError` for when the user passes wrong data type for either of the 3 properties; should I change it to `TypeError` instead?

Reply:
Because on line 42 it is expected that an exception is raised in case if any foreign key relationships is missing.
If you expect a particular exception to happen and you need to handle it in a specific manner, then you handle it.

But here the exception handling serves no purpose, doesn't it?

Reply:
Correct.